src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -511,8 +511,6 @@ public static int BinarySearch(Array array, int index, int length, object? value
             {
                 CorElementType et = array.GetCorElementTypeOfElementType();
                 if (et.IsPrimitiveType()
-                    // IntPtr/UIntPtr does not implement IComparable
-                    && (et != CorElementType.ELEMENT_TYPE_I) && (et != CorElementType.ELEMENT_TYPE_U))
```

Comment:
I think the handling for `ELEMENT_TYPE_I` and `ELEMENT_TYPE_U` is missing just below.

We likely want to have a test covering these methods as well.